core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -137,7 +137,9 @@ object TestUtils extends Logging {
     val parentFile = new File(parent)
     parentFile.mkdirs()
 
-    JTestUtils.tempDirectory(parentFile.toPath, null)
+    val result = JTestUtils.tempDirectory(parentFile.toPath, null)
```

Comment:
Note that `JTestUtils.tempDirectory` is overloaded with `parentFile` possibly being null. In those cases, the cleanup works as intended.

This is however a special case where `parentFile != null`, because the test create a logdir in a "relative location". `JTestUtils.tempDirectory` does handle the removal of everything it creates under `parentFile` but it doesn't remove the `parentFile` though.

I feel it's semantically better for the function that creates `parentFile` to handle its deletion rather than delegating it to `JTestUtils.tempDirectory` since it might not be very obvious to callers of `JTestUtils.tempDirectory` otherwise.